---
title: "20200624_GO_DG_Subi_genes_analysis"
author: "Skim"
date: '2020 6 24 '
output: 
  html_document:
    df_print: paged
    toc: true
    toc_depth: 4
    toc_float: false
    code_folding: hide
---

In the report, I checked GO analysis (AD vs ADp40KO) terms genes and how they are differentially expressed among samples. I looked at by genotype, sample and hierarchical clustering. 

```{r warning=FALSE, message=FALSE}
#.libPaths(c("/home/skim/R/usr_lib", .libPaths()))
library(Seurat, lib.loc = "/data/rajewsky/shared_libs/R/")
library(ggplot2)
library(gridExtra)
library(grid)
library(ggrepel)
library(DT)
library(biomaRt)
library(dplyr)
library(viridis)
library(tidyr)
library(RColorBrewer)
library(scales)
```

## 1. Dentate Gyrus regions (cluster 0, 10)

AD MF  
1) calmodulin binding (Cacna1c, Syt1, Pde1a, Slc8a1)   
2) calcium ion transmembrane transporter (Fam155a, Il1rapl1, Cacna1c, Slc24a3, Slc8a1) = calcium channel activity    
3) sodium ion transmembrane transporter (Asic2, Slc24a3, Grik4, Slc8a1)  

AD BP  
1) regulation of presynapse assembly (Il1rapl1, Mdga2, Gpc4)  
2) homophilic cell adhesion via plasma (Clstn2, Kirrel3, Cdh12, Dscam)  
3) heterophilic cell-cell adhesion (Il1rapl1, Tenm1, Tenm4)  
4) positive regulation of synapse assembly (Clstn2, Asic2, Il1rapl1)  
5) synaptic transmission, glutamatergic (Syt1, Tnr, Grik4)  

ADp40KO MF  
1) protein serine/threonine kinase inhibitor (Spred2, Spred1)  
2) transmembrane receptor protein tyrosine (Epha5, Epha4)  
3) cadherin binding (Cdh9, Ctnna3)  

ADp40KO BP  
1) axon guidance (Epha5, Epha4, Etv1, Ntng1, Sema6d, Unc5d)  
2) ephrin receptor signaling pathway (Epha5, Epha4)  
3) regulation of protein deacetylation (Prkd1, Spred2, Spred1)  
4) regulation of neuron projection development (Ahi1, Epha4, Prkd1, Syne1, Sema6d)  

```{r warning=FALSE, message=FALSE}
load(file = "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200221_figure1_for_meeting_report_Seurat_object.Robj")
```

```{r warning=FALSE, message=FALSE, fig.width=8.5, fig.height=5}
z_score_DE_genes <- function(Seurat_object, list_genes){
  for(i in c(1:9)){
  if(i == 1){
    cell_avg.df <- as.data.frame(Matrix::rowMeans(GetAssayData(object = subset(Seurat_object, subset = gemgroup %in% i), slot = 'data')))
  }
  else{
    tempcell_avg.df <- as.data.frame(Matrix::rowMeans(GetAssayData(object = subset(Seurat_object, subset = gemgroup %in% i), slot = 'data')))
  cell_avg.df <- cbind(cell_avg.df, tempcell_avg.df)
  }
}
#print("1st")
cell_avg.df$gene <- rownames(cell_avg.df)
#print("2nd")
cell_avg.df <- cell_avg.df[cell_avg.df$gene %in% list_genes, ]
#print("3rd")
cell_avg.df$gene <- NULL

# calculate z-score (high)
cell_avg_z_score.df <- as.data.frame(t(apply(cell_avg.df, 1, function(x) (x - mean(x)) / sd(x))))
return(cell_avg_z_score.df)
} 

z_score_DE_genes_by_all <- function(Seurat_object, list_genes){
  for(i in c("Ctrl", "AD", "ADp40KO")){
  if(i == "Ctrl"){
    cell_avg.df <- as.data.frame(Matrix::rowMeans(GetAssayData(object = subset(Seurat_object, subset = sample %in% i), slot = 'data')))
  }
  else{
    tempcell_avg.df <- as.data.frame(Matrix::rowMeans(GetAssayData(object = subset(Seurat_object, subset = sample %in% i), slot = 'data')))
  cell_avg.df <- cbind(cell_avg.df, tempcell_avg.df)
  }
}


cell_avg.df$gene <- rownames(cell_avg.df)

cell_avg.df <- cell_avg.df[cell_avg.df$gene %in% list_genes, ]

cell_avg.df$gene <- NULL

# calculate z-score (high)
cell_avg_z_score.df <- as.data.frame(t(apply(cell_avg.df, 1, function(x) (x - mean(x)) / sd(x))))
return(cell_avg_z_score.df)
}
```

```{r warning=FALSE, message=FALSE, fig.width=8.5, fig.height=5}
data9set_DG.SO <- subset(data9set_cleaned.SO, subset = seurat_clusters %in% c(0, 10))
```

### 1-1. AD MF in DG

```{r warning=FALSE, message=FALSE}
calmodulin_binding <- c("Cacna1c", "Syt1", "Pde1a", "Slc8a1")   
calcium_ion_transmembrane_transporter <- c("Fam155a", "Il1rapl1", "Cacna1c", "Slc24a3", "Slc8a1")   
sodium_ion_transmembrane_transporter <- c("Asic2", "Slc24a3", "Grik4", "Slc8a1")

genes <- c(calmodulin_binding, calcium_ion_transmembrane_transporter, sodium_ion_transmembrane_transporter) %>% unique
  
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_DG.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% calmodulin_binding, "calmodulin\nbinding", ifelse(z_score$gene %in% setdiff(sodium_ion_transmembrane_transporter, calmodulin_binding), "sodium\nion\ntransmembrane\ntransporter", "calcium\nion\ntransmembrane\ntransporter"))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_DG.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% calmodulin_binding, "calmodulin\nbinding", ifelse(z_score$gene %in% setdiff(sodium_ion_transmembrane_transporter, calmodulin_binding), "sodium\nion\ntransmembrane\ntransporter", "calcium\nion\ntransmembrane\ntransporter"))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
VlnPlot(data9set_DG.SO, features = "Slc8a1", group.by = "gemgroup", pt.size = 0.01) + theme(legend.position = "none")
```

### 1-2. AD BP in DG

```{r warning=FALSE, message=FALSE}
regulation_of_presynapse_assembly <- c("Il1rapl1", "Mdga2", "Gpc4")  
homophilic_cell_adhesion_via_plasma <- c("Clstn2", "Kirrel3", "Cdh12", "Dscam")  
heterophilic_cell_cell_adhesion <- c("Il1rapl1", "Tenm1", "Tenm4")  
positive_regulation_of_synapse_assembly <- c("Clstn2", "Asic2", "Il1rapl1")  
synaptic_transmission_glutamatergic <- c("Syt1", "Tnr", "Grik4") 

genes <- c(regulation_of_presynapse_assembly, homophilic_cell_adhesion_via_plasma, heterophilic_cell_cell_adhesion,
           positive_regulation_of_synapse_assembly, synaptic_transmission_glutamatergic) %>% unique
  
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_DG.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% regulation_of_presynapse_assembly, "regulation\nof\npresynapse\nassembly", 
                     ifelse(z_score$gene %in% setdiff(homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly), "homophilic\ncell\nadhesion\nvia\nplasma", 
                            ifelse(z_score$gene %in% setdiff(heterophilic_cell_cell_adhesion, c(homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly)), "heterophilic\ncell\ncell\nadhesion", 
                                   ifelse(z_score$gene %in% setdiff(positive_regulation_of_synapse_assembly, c(heterophilic_cell_cell_adhesion, homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly)), "positive\nregulation\nof\nsynapse\nassembly", "synaptic\ntransmission\nglutamatergic"))))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_DG.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% regulation_of_presynapse_assembly, "regulation\nof\npresynapse\nassembly", 
                     ifelse(z_score$gene %in% setdiff(homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly), "homophilic\ncell\nadhesion\nvia\nplasma", 
                            ifelse(z_score$gene %in% setdiff(heterophilic_cell_cell_adhesion, c(homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly)), "heterophilic\ncell\ncell\nadhesion", 
                                   ifelse(z_score$gene %in% setdiff(positive_regulation_of_synapse_assembly, c(heterophilic_cell_cell_adhesion, homophilic_cell_adhesion_via_plasma, regulation_of_presynapse_assembly)), "positive\nregulation\nof\nsynapse\nassembly", "synaptic\ntransmission\nglutamatergic"))))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
VlnPlot(data9set_DG.SO, features = "Clstn2", group.by = "gemgroup", pt.size = 0.01) + theme(legend.position = "none")
```

### 1-3. ADp40KO MF in DG

```{r warning=FALSE, message=FALSE}
protein_serine_threonine_kinase_inhibitor <- c("Spred2", "Spred1")  
transmembrane_receptor_protein_tyrosine <- c("Epha5", "Epha4")  
cadherin_binding <- c("Cdh9", "Ctnna3")  

genes <- c(protein_serine_threonine_kinase_inhibitor, transmembrane_receptor_protein_tyrosine, cadherin_binding) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_DG.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% protein_serine_threonine_kinase_inhibitor, "protein\nserine\nthreonine\nkinase\ninhibitor", ifelse(z_score$gene %in% setdiff(transmembrane_receptor_protein_tyrosine, protein_serine_threonine_kinase_inhibitor), "transmembrane\nreceptor\nprotein\ntyrosine", "cadherin\nbinding"))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=4}
z_score <- z_score_DE_genes(data9set_DG.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% protein_serine_threonine_kinase_inhibitor, "protein\nserine\nthreonine\nkinase\ninhibitor", ifelse(z_score$gene %in% setdiff(transmembrane_receptor_protein_tyrosine, protein_serine_threonine_kinase_inhibitor), "transmembrane\nreceptor\nprotein\ntyrosine", "cadherin\nbinding"))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 1-4. ADp40KO BP in DG

```{r warning=FALSE, message=FALSE}
axon_guidance <- c("Epha5", "Epha4", "Etv1", "Ntng1", "Sema6d", "Unc5d")  
ephrin_receptor_signaling_pathway <- c("Epha5", "Epha4")  
regulation_of_protein_deacetylation <- c("Prkd1", "Spred2", "Spred1")  
regulation_of_neuron_projection_development <- c("Ahi1", "Epha4", "Prkd1", "Syne1", "Sema6d")   

genes <- c(axon_guidance, ephrin_receptor_signaling_pathway, regulation_of_protein_deacetylation, regulation_of_neuron_projection_development) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_DG.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% ephrin_receptor_signaling_pathway, "ephrin\nreceptor\nsignaling\npathway", 
                     ifelse(z_score$gene %in% setdiff(regulation_of_protein_deacetylation, ephrin_receptor_signaling_pathway), "regulation\nof\nprotein\ndeacetylation", 
                            ifelse(z_score$gene %in% setdiff(axon_guidance, c(regulation_of_protein_deacetylation, ephrin_receptor_signaling_pathway)), "axon\nguidance", "regulation\nof\nneuron\nprojection\ndevelopment")))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
z_score <- z_score_DE_genes(data9set_DG.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% ephrin_receptor_signaling_pathway, "ephrin\nreceptor\nsignaling\npathway", 
                     ifelse(z_score$gene %in% setdiff(regulation_of_protein_deacetylation, ephrin_receptor_signaling_pathway), "regulation\nof\nprotein\ndeacetylation", 
                            ifelse(z_score$gene %in% setdiff(axon_guidance, c(regulation_of_protein_deacetylation, ephrin_receptor_signaling_pathway)), "axon\nguidance", "regulation\nof\nneuron\nprojection\ndevelopment")))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 1-5. cluster 0 cell proportion

sample 5 showed much more cells proportion than other samples

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}
plot.data <- data9set_cleaned.SO@meta.data %>%
    dplyr::select(sample, gemgroup, cluster = seurat_clusters) %>%
    mutate(cluster = cluster) %>%
    group_by(gemgroup, cluster, sample) %>%
  summarise(count = n()) %>%
  group_by(gemgroup) %>%
    mutate(gemgroup_total = sum(count)) %>%
    mutate(cell_prop = count / gemgroup_total)

plot.data$gemgroup <- factor(plot.data$gemgroup, levels = c(1,4,7,2,5,8,3,6,9))

ggplot(plot.data[plot.data$cluster %in% c(0),], aes(x=gemgroup, y=cell_prop, fill = sample, color=sample)) +
  geom_point(position=position_dodge(0.8), size = 2) + geom_text_repel(aes(label = gemgroup), 
                    size = 6, color = "black", force =  10)
```

### 1-6. cluster 10 cell proportion

sample 5 showed much less cells proportion than other samples

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}

ggplot(plot.data[plot.data$cluster %in% c(10),], aes(x=gemgroup, y=cell_prop, fill = sample, color=sample)) +
  geom_point(position=position_dodge(0.8), size = 2) + geom_text_repel(aes(label = gemgroup), 
                    size = 6, color = "black", force =  10)
```

## 2. Dentate Gyrus regions (cluster 11, 14, 15, 17, 19, 20)

```{r warning=FALSE, message=FALSE, fig.width=8.5, fig.height=5}
data9set_subi.SO <- subset(data9set_cleaned.SO, subset = seurat_clusters %in% c(11, 14, 15, 17, 19, 20))
```

AD MF  
1) glutamate receptor activity (Gria1, Grin2b, Grin2a, Grid2, Gria4)   
2) calcium ion binding (Dgkb, Cdh4, Fstl4, Caln1, Plcb1, Cacna1e, Necab1, Kcnip4, Slit3, Lrp1b, Scube1, Cdh9, Slit2, Cdh13)  
3) cadherin binding (Cdh4, Ctnna3, Ptpro, Cdh9, Cdh13)  
4) ion channel binding (Kcnc2, Cacna1c, Hap1, Kcnq3, Camk2d, Kcnd3)  
5) voltage-gated cation channel activity (Grin2b, Cacna2d3, Kcnc2, Oprm1, Cacna1c, Grin2a, Cacna1e, Kcnq3, Kcnt2, Kcnd3)

AD BP  
1) receptor clustering (Grin2b, Sorbs2, Glra3, Dlg2, Lrrc7, Lrrtm4, Arhgef9)  
2) heterophilic cell-cell adhesion (Cdh4, Tenm4, Grid2, Alcam, Tenm1)  
3) regulation of postsynapse organization (Dgkb, Grin2b, Sorbs2, Grid2, Ntrk3, Lrfn2, Arhgef9)  
4) excitatory postsynaptic potential (Grin2b, Oprm1, Grin2a, Celf4, Grid2, Mef2c)  
5) Rac protein signal transduction (Camk2d, Auts2, Elmo1, Cdh13)  
6) negative chemotaxis (Unc5c, Slit3, Slit2, Lrtm1)  

ADp40KO MF  
1) transmembrane receptor protein tyrosine  (Nrp1, Kit, Ephb2, Nrp2, Alk, Epha3)  
2) calcium ion binding (Nrxn1, Kcnip1, Crtac1, Ldlr, Sulf2, Fat3, Cdh11, Astn2, Atp2b2, Fstl5, Dst, Atp2a2, Syt7, Spock2, Man1a)   
3) microtubule binding (Map1b, Fmn1, Skap1, Kif5a, Dpysl2, Dst, Mtcl1, Kif5c)  
4) lipid binding (Map1b, Epb41, Dab1, Thy1, Arap2, Stard5, Pitpnc1, Osbpl6, Ppard, Nr3c2, Psap, Syt7, Syt11, Grk5, Zfyve9)  
5) heparin binding (Nav2, Nrp1, Ncam1, Hsd17b12, Nrp2)

ADp40KO BP  
1) plasma membrane organization (Xkr6, Ano4, Atp2a2, Psap, Syt7, Syt11)  
2) negative regulation of axonogenesis (Sema6d, Dab1, Thy1, Nrp1, Ephb2)  
3) synaptic vesicle transport (Kif5a, Nr3c2, Dpysl2, Atp2a2, Sv2a, Syt7, Grm8, Kif5c, Syt11)  
4) axon guidance (Sema6d, Nrp1, Kif5a, Ncam1, Dpysl2, Cntn6, Ephb2, Kif5c, Nrp2, Epha3)  
5) positive regulation of neuron differentiation (Map1b, Vwc2l, Nedd4l, Dab1, Nrp1, Tcf4, Spag9, Afdn, Serpini1, Pak1, Alk, Etv5, Epha3)  
6) dendrite development (Map1b, Fmn1, Hecw2, Nedd4l, Dab1, Fat3, Nrp1, Ephb2, Afdn, Pak1, Alk)

### 2-1. AD MF in Subiculum

```{r warning=FALSE, message=FALSE}
glutamate_receptor_activity <- c("Gria1", "Grin2b", "Grin2a", "Grid2", "Gria4")   
calcium_ion_binding <- c("Dgkb", "Cdh4", "Fstl4", "Caln1", "Plcb1", "Cacna1e", "Necab1", "Kcnip4", "Slit3", "Lrp1b", "Scube1", "Cdh9", "Slit2", "Cdh13")  
cadherin_binding <- c("Cdh4", "Ctnna3", "Ptpro", "Cdh9", "Cdh13")  
ion_channel_binding <- c("Kcnc2", "Cacna1c", "Hap1", "Kcnq3", "Camk2d", "Kcnd3")  
voltage_gated_cation_channel_activity <- c("Grin2b", "Cacna2d3", "Kcnc2", "Oprm1", "Cacna1c", "Grin2a", "Cacna1e", "Kcnq3", "Kcnt2", "Kcnd3") 

genes <- c(cadherin_binding, glutamate_receptor_activity, ion_channel_binding, calcium_ion_binding, voltage_gated_cation_channel_activity) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_subi.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% cadherin_binding, "cadherin\nbinding", 
                     ifelse(z_score$gene %in% setdiff(glutamate_receptor_activity, cadherin_binding), "glutamate\nreceptor\nactivity", 
                            ifelse(z_score$gene %in% setdiff(ion_channel_binding, c(glutamate_receptor_activity, cadherin_binding)), "ion\nchannel\nbinding", 
                                   ifelse(z_score$gene %in% setdiff(calcium_ion_binding, c(ion_channel_binding, glutamate_receptor_activity, cadherin_binding)), "calcium\nion\nbinding", "voltage\ngated\ncation\nchannel\nactivity"))))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_subi.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% cadherin_binding, "cadherin\nbinding", 
                     ifelse(z_score$gene %in% setdiff(glutamate_receptor_activity, cadherin_binding), "glutamate\nreceptor\nactivity", 
                            ifelse(z_score$gene %in% setdiff(ion_channel_binding, c(glutamate_receptor_activity, cadherin_binding)), "ion\nchannel\nbinding", 
                                   ifelse(z_score$gene %in% setdiff(calcium_ion_binding, c(ion_channel_binding, glutamate_receptor_activity, cadherin_binding)), "calcium\nion\nbinding", "voltage\ngated\ncation\nchannel\nactivity"))))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 2-2. AD BP in Subiculum

```{r warning=FALSE, message=FALSE}
receptor_clustering <- c("Grin2b", "Sorbs2", "Glra3", "Dlg2", "Lrrc7", "Lrrtm4", "Arhgef9")  
heterophilic_cell_cell_adhesion <- c("Cdh4", "Tenm4", "Grid2", "Alcam", "Tenm1")  
regulation_of_postsynapse_organization <- c("Dgkb", "Grin2b", "Sorbs2", "Grid2", "Ntrk3", "Lrfn2", "Arhgef9")  
excitatory_postsynaptic_potential <- c("Grin2b", "Oprm1", "Grin2a", "Celf4", "Grid2", "Mef2c")  
Rac_protein_signal_transduction <- c("Camk2d", "Auts2", "Elmo1", "Cdh13")  
negative_chemotaxis <- c("Unc5c", "Slit3", "Slit2", "Lrtm1") 

genes <- c(negative_chemotaxis, Rac_protein_signal_transduction, heterophilic_cell_cell_adhesion, receptor_clustering, excitatory_postsynaptic_potential, regulation_of_postsynapse_organization) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_subi.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% negative_chemotaxis, "negative\nchemotaxis", 
                     ifelse(z_score$gene %in% setdiff(Rac_protein_signal_transduction, negative_chemotaxis), "Rac\nprotein\nsignal\ntransduction", 
                            ifelse(z_score$gene %in% setdiff(heterophilic_cell_cell_adhesion, c(Rac_protein_signal_transduction, negative_chemotaxis)), "heterophilic\ncell\ncell\nadhesion", 
                                   ifelse(z_score$gene %in% setdiff(receptor_clustering, c(heterophilic_cell_cell_adhesion, Rac_protein_signal_transduction, negative_chemotaxis)), "receptor\nclustering", 
                                          ifelse(z_score$gene %in% setdiff(excitatory_postsynaptic_potential, c(receptor_clustering, heterophilic_cell_cell_adhesion, Rac_protein_signal_transduction, negative_chemotaxis)), "excitatory\npostsynaptic\npotential", "regulation\nof\npostsynapse\norganization")))))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```


```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_subi.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% negative_chemotaxis, "negative\nchemotaxis", 
                     ifelse(z_score$gene %in% setdiff(Rac_protein_signal_transduction, negative_chemotaxis), "Rac\nprotein\nsignal\ntransduction", 
                            ifelse(z_score$gene %in% setdiff(heterophilic_cell_cell_adhesion, c(Rac_protein_signal_transduction, negative_chemotaxis)), "heterophilic\ncell\ncell\nadhesion", 
                                   ifelse(z_score$gene %in% setdiff(receptor_clustering, c(heterophilic_cell_cell_adhesion, Rac_protein_signal_transduction, negative_chemotaxis)), "receptor\nclustering", 
                                          ifelse(z_score$gene %in% setdiff(excitatory_postsynaptic_potential, c(receptor_clustering, heterophilic_cell_cell_adhesion, Rac_protein_signal_transduction, negative_chemotaxis)), "excitatory\npostsynaptic\npotential", "regulation\nof\npostsynapse\norganization")))))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 2-3. ADp40KO MF in Subiculum

```{r warning=FALSE, message=FALSE}
transmembrane_receptor_protein_tyrosine <- c("Nrp1", "Kit", "Ephb2", "Nrp2", "Alk", "Epha3")  
calcium_ion_binding <- c("Nrxn1", "Kcnip1", "Crtac1", "Ldlr", "Sulf2", "Fat3", "Cdh11", "Astn2", "Atp2b2", "Fstl5", "Dst", "Atp2a2", "Syt7", "Spock2", "Man1a")   
microtubule_binding <- c("Map1b", "Fmn1", "Skap1", "Kif5a", "Dpysl2", "Dst", "Mtcl1", "Kif5c")  
lipid_binding <- c("Map1b", "Epb41", "Dab1", "Thy1", "Arap2", "Stard5", "Pitpnc1", "Osbpl6", "Ppard", "Nr3c2", "Psap", "Syt7", "Syt11", "Grk5", "Zfyve9")  
heparin_binding <- c("Nav2", "Nrp1", "Ncam1", "Hsd17b12", "Nrp2")

genes <- c(transmembrane_receptor_protein_tyrosine, microtubule_binding, heparin_binding, calcium_ion_binding, lipid_binding) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_subi.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% transmembrane_receptor_protein_tyrosine, "transmembrane\nreceptor\nprotein\ntyrosine", 
                     ifelse(z_score$gene %in% setdiff(microtubule_binding, transmembrane_receptor_protein_tyrosine), "microtubule\nbinding", 
                            ifelse(z_score$gene %in% setdiff(heparin_binding, c(microtubule_binding, transmembrane_receptor_protein_tyrosine)), "heparin\nbinding", 
                                   ifelse(z_score$gene %in% setdiff(calcium_ion_binding, c(heparin_binding, microtubule_binding, transmembrane_receptor_protein_tyrosine)), "calcium\nion\nbinding", "lipid\nbinding"))))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_subi.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% transmembrane_receptor_protein_tyrosine, "transmembrane\nreceptor\nprotein\ntyrosine", 
                     ifelse(z_score$gene %in% setdiff(microtubule_binding, transmembrane_receptor_protein_tyrosine), "microtubule\nbinding", 
                            ifelse(z_score$gene %in% setdiff(heparin_binding, c(microtubule_binding, transmembrane_receptor_protein_tyrosine)), "heparin\nbinding", 
                                   ifelse(z_score$gene %in% setdiff(calcium_ion_binding, c(heparin_binding, microtubule_binding, transmembrane_receptor_protein_tyrosine)), "calcium\nion\nbinding", "lipid\nbinding"))))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 2-4. ADp40KO BP in Subiculum

```{r warning=FALSE, message=FALSE}
plasma_membrane_organization <- c("Xkr6", "Ano4", "Atp2a2", "Psap", "Syt7", "Syt11")  
negative_regulation_of_axonogenesis <- c("Sema6d", "Dab1", "Thy1", "Nrp1", "Ephb2")  
synaptic_vesicle_transport <- c("Kif5a", "Nr3c2", "Dpysl2", "Atp2a2", "Sv2a", "Syt7", "Grm8", "Kif5c", "Syt11")  
axon_guidance <- c("Sema6d", "Nrp1", "Kif5a", "Ncam1", "Dpysl2", "Cntn6", "Ephb2", "Kif5c", "Nrp2", "Epha3")  
positive_regulation_of_neuron_differentiation <- c("Map1b", "Vwc2l", "Nedd4l", "Dab1", "Nrp1", "Tcf4", "Spag9", "Afdn", "Serpini1", "Pak1", "Alk", "Etv5", "Epha3")  
dendrite_development <- c("Map1b", "Fmn1", "Hecw2", "Nedd4l", "Dab1", "Fat3", "Nrp1", "Ephb2", "Afdn", "Pak1", "Alk")

genes <- c(plasma_membrane_organization, negative_regulation_of_axonogenesis, synaptic_vesicle_transport, axon_guidance, positive_regulation_of_neuron_differentiation, dendrite_development) %>% unique
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=3}
z_score <- z_score_DE_genes_by_all(data9set_subi.SO, genes)
colnames(z_score) <- c("WT", "AD", "AD.p40KO")

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% plasma_membrane_organization, "plasma\nmembrane\norganization", 
                     ifelse(z_score$gene %in% setdiff(negative_regulation_of_axonogenesis, plasma_membrane_organization), "negative\nregulation\nof\naxonogenesis", 
                            ifelse(z_score$gene %in% setdiff(synaptic_vesicle_transport, c(negative_regulation_of_axonogenesis, plasma_membrane_organization)), "synaptic\nvesicle\ntransport", 
                                   ifelse(z_score$gene %in% setdiff(axon_guidance, c(synaptic_vesicle_transport, negative_regulation_of_axonogenesis, plasma_membrane_organization)), "axon\nguidance", 
                                          ifelse(z_score$gene %in% setdiff(positive_regulation_of_neuron_differentiation, c(axon_guidance, synaptic_vesicle_transport, negative_regulation_of_axonogenesis, plasma_membrane_organization)), "positive\nregulation\nof\nneuron\ndifferentiation", "dendrite\ndevelopment")))))

z_score2 <- gather(z_score, sample, z_score, WT:AD.p40KO)

z_score2$sample <- factor(z_score2$sample, levels = c("WT","AD.p40KO","AD"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=4}
z_score <- z_score_DE_genes(data9set_subi.SO, genes)
colnames(z_score) <- c("WT_1", "AD.p40KO_1", "AD_1", "WT_2", "AD.p40KO_2", "AD_2", "WT_3", "AD.p40KO_3", "AD_3")

mat <- z_score

z_score$gene <- rownames(z_score)

z_score$GO <- ifelse(z_score$gene %in% plasma_membrane_organization, "plasma\nmembrane\norganization", 
                     ifelse(z_score$gene %in% setdiff(negative_regulation_of_axonogenesis, plasma_membrane_organization), "negative\nregulation\nof\naxonogenesis", 
                            ifelse(z_score$gene %in% setdiff(synaptic_vesicle_transport, c(negative_regulation_of_axonogenesis, plasma_membrane_organization)), "synaptic\nvesicle\ntransport", 
                                   ifelse(z_score$gene %in% setdiff(axon_guidance, c(synaptic_vesicle_transport, negative_regulation_of_axonogenesis, plasma_membrane_organization)), "axon\nguidance", 
                                          ifelse(z_score$gene %in% setdiff(positive_regulation_of_neuron_differentiation, c(axon_guidance, synaptic_vesicle_transport, negative_regulation_of_axonogenesis, plasma_membrane_organization)), "positive\nregulation\nof\nneuron\ndifferentiation", "dendrite\ndevelopment")))))

z_score2 <- gather(z_score, sample, z_score, WT_1:AD_3)

z_score2$sample <- factor(z_score2$sample, levels = c("WT_1", "WT_2", "WT_3", "AD.p40KO_1", "AD.p40KO_2", "AD.p40KO_3", "AD_1", "AD_2","AD_3"))

cols <- rev(brewer.pal(11, 'RdYlBu'))

ggplot(data = z_score2, mapping = aes(x = sample, y = gene, fill = z_score)) +
  geom_tile() + scale_y_discrete(position = "right")  + coord_flip() +  theme_classic() + theme(plot.title = element_text(size = 15, hjust = 0.5), 
                                        axis.title.y = element_blank(),
                                         axis.title.x = element_blank(),
                                         axis.text.x = element_text(angle = 90, vjust = 11.5, size = 12, color = "black"),
                                         axis.text.y = element_text(size = 12, color = "black"),
                                         axis.line = element_line(color = "white"),
                                         axis.ticks = element_line(color = "white"),
                                         legend.title = element_blank(),
                                         legend.text = element_text(size = 11, color = "black"),
                                         panel.spacing = unit(0, "mm"), 
                                        panel.margin= unit(c(-1), "mm"), 
                                        plot.margin = margin(0, 0, 0, 0, "cm"),
                                        strip.background = element_blank(),
                                        strip.text.x = element_text(size = 10)) + 
  #scale_fill_gradient2(low="red", high='blue', mid="white", midpoint = 0) +  
  scale_fill_gradientn(colours = cols, limits = c(-2, 2), oob=squish) + 
  facet_grid(~GO, scales = "free_x", space = "free_x", switch = 'x')
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
mat <- as.matrix(t(mat))
heatmap(mat, scale = "none", col =  cols, margins = c(10,1))
```

### 2-5. cluster 11 cell proportion

sample 5 showed much less cells proportion than other samples

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}

ggplot(plot.data[plot.data$cluster %in% c(11),], aes(x=gemgroup, y=cell_prop, fill = sample, color=sample)) +
  geom_point(position=position_dodge(0.8), size = 2) + geom_text_repel(aes(label = gemgroup), 
                    size = 6, color = "black", force =  10)
```

### 2-6. cluster 15 cell proportion

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}

ggplot(plot.data[plot.data$cluster %in% c(15),], aes(x=gemgroup, y=cell_prop, fill = sample, color=sample)) +
  geom_point(position=position_dodge(0.8), size = 2) + geom_text_repel(aes(label = gemgroup), 
                    size = 6, color = "black", force =  10)
```

```{r warning=FALSE, message=FALSE, fig.width=8, fig.height=4}
sessionInfo()
```