---
title: "lncRNA_miRNA_feedback meeting"
author: "Skim"
date: '2020 6 3 '
output: 
  html_document:
    df_print: paged
    toc: true
    toc_depth: 4
    toc_float: false
    code_folding: hide
---

```{r warning=FALSE, message=FALSE}
#.libPaths(c("/home/skim/R/usr_lib", .libPaths()))
#.libPaths(c(.libPaths()))
#library(Seurat, lib.loc="/data/rajewsky/shared_libs/R/")
library(Seurat)
library(tidyverse)
library(dplyr)
library(stringr)
library(ggplot2)
library(gridExtra)
library(xlsx)
library(readxl)
library(topGO)
library(ggrepel)
```

### 1. Calculate CPM of WT samples from snRNA-seq

```{r warning=FALSE, message=FALSE}
# load(file="/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/Seurat_object/20200501_Seurat_object_Ctrl_Robj")
# 
# data9set_cleaned_Ctrl.df <- as.data.frame(data9set_cleaned_Ctrl.SO@assays$RNA@counts)
# cpm <- apply(data9set_cleaned_Ctrl.df, 2, function(x) (x/sum(x))*1000000)
# cpm.df <- rowMeans(cpm) %>% as.data.frame
# 
# # add pseudocount 1
# cpm.df <- cpm.df + 1
# 
# # log2 
# cpm_log2.df <- as.data.frame(lapply(cpm.df, log2), row.names = rownames(cpm.df))
```

### 2. Calculate TPM of WT samples from bulk data

```{r warning=FALSE, message=FALSE}
# bulk data load
# DGEm <- read.csv("/data/rajewsky/home/skim/Microglia_Heppner/Bulk_transcriptome/R_Scripts/20200311_Bulk_Analysis_Modified_Final_DF.csv", row.names =1, sep = ",", check.names = FALSE)

###################
# load gene length
###################
# Below I create gene_length DF to change ENSEMBL to mgi_symbol

# gene_length.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/Bulk_transcriptome/GTFtools_0.6.9/GRCm38_M21.isoform.length.txt", header = TRUE)
# 
# mart <- useDataset("mmusculus_gene_ensembl", mart = useMart("ensembl"))
# 
# test_mouse_genes <- getBM(attributes = c("ensembl_gene_id_version", "mgi_symbol"),
#                     filters = 'ensembl_gene_id_version',
#                     values = gene_length.df$gene,
#                     mart = mart)
# 
# colnames(test_mouse_genes) <- c("gene", "mgi_symbol")
# 
# gene_length.df2 <- inner_join(gene_length.df, test_mouse_genes, by = "gene")
# 
# write.table(gene_length.df2, file = "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200603_lncRNA_CPM_bulk_TPM_isoform.length.txt")

# # load gene length
# gene_length.df2 <- read.table(file = "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200603_lncRNA_CPM_bulk_TPM_isoform.length.txt")
# 
# gene_length_median.df <- gene_length.df2 %>% group_by(mgi_symbol) %>% dplyr::summarise(Median=median(length))
# 
# colnames(gene_length_median.df) <- c("mgi_symbol", "length")
# 
# #table(duplicated(gene_length_median.df$mgi_symbol))
# 
# DGEm$mgi_symbol <- rownames(DGEm)
# 
# DGEm_length <- inner_join(DGEm, gene_length_median.df, by = c("mgi_symbol"))
# 
# rownames(DGEm_length) <- DGEm_length$mgi_symbol
# DGEm_length$mgi_symbol <- NULL
# 
# DGEm_length_info.df <- data.frame(length = DGEm_length$length, row.names = rownames(DGEm_length))
# DGEm_length$length <-NULL
# 
# ########
# # TPM
# ########
# tpm <- function(counts, lengths) {
#   rate <- counts / lengths # RPK
#   rate.df <- rate / sum(rate) * 1e6
#   return(rate.df)
# }
# 
# DGEm_length.df <- apply(DGEm_length, 2, function(x) tpm(x, DGEm_length_info.df$length)) %>% as.data.frame
# 
# #. add pseudocount 1
# DGEm_length.df <- DGEm_length.df + 1
# 
# # log2 
# DGEm_length_log2.df <- as.data.frame(lapply(DGEm_length.df, log2), row.names = rownames(DGEm_length.df))

```

### 3. merge CPM and TPM of WT data

Both data have 23993 genes expression 

```{r warning=FALSE, message=FALSE}
# DGEm_WT <- data.frame(TPM = rowMeans(DGEm_length_log2.df[,c(1,3,7)]), row.names = rownames(DGEm_length_log2.df))
# DGEm_WT$gene <- rownames(DGEm_WT)
# 
# colnames(cpm_log2.df) <- "CPM"
# cpm_log2.df$gene <- rownames(cpm_log2.df)
# 
# WT_TPM_CPM <- inner_join(DGEm_WT, cpm_log2.df, by = "gene")
# 
# write.table(WT_TPM_CPM, file = "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200603_lncRNA_CPM_bulk_TPM_WT_TPM_CPM.txt")

WT_TPM_CPM <- read.table(file = "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200603_lncRNA_CPM_bulk_TPM_WT_TPM_CPM.txt")

```

### 4. Extract only lncRNAs

```{r warning=FALSE, message=FALSE}
# load biotype
seurat_mouse_genes <- read.table(file = "/data/rajewsky/home/skim/Microglia_Heppner/Bulk_transcriptome/R_Scripts/20200312_Bulk_Correlation_mouse_genes_bio_type_biomaRt.txt")

# narrow down biotype
seurat_mouse_genes$gene_biotype_global <- ifelse(seurat_mouse_genes$gene_biotype %in% c("IG_V_gene", "IG_C_gene"), "IG_gene",
                                                 ifelse(seurat_mouse_genes$gene_biotype %in% c("pseudogene",
                                                                                               "unprocessed_pseudogene",
                                                                                               "processed_pseudogene",
                                                                                               "transcribed_unprocessed_pseudogene",
                                                                                               "transcribed_processed_pseudogene",
                                                                                               "unitary_pseudogene", 
                                                                                               "transcribed_unitary_pseudogene",
                                                                                               "polymorphic_pseudogene", 
                                                                                               "TR_V_pseudogene"), "pseudogene",
                                                 ifelse(seurat_mouse_genes$gene_biotype %in% c("TR_V_gene", "TR_C_gene"), "TR_gene",
                                                 ifelse(seurat_mouse_genes$gene_biotype %in% c("protein_coding"), "protein_coding",
                                                 ifelse(seurat_mouse_genes$gene_biotype %in% c("TEC"), "TEC",
                                                 ifelse(seurat_mouse_genes$gene_biotype %in% c("lncRNA"), "lncRNA", "ncRNA"))))))
```

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height= 5}
# only lncRNAs
lncRNA_WT_TPM_CPM.df <- WT_TPM_CPM[WT_TPM_CPM$gene %in% seurat_mouse_genes[seurat_mouse_genes$gene_biotype_global %in% "lncRNA", ]$mgi_symbol, ]

lncRNA_WT_TPM_CPM_sub.df <- lncRNA_WT_TPM_CPM.df[lncRNA_WT_TPM_CPM.df$gene %in% c("Malat1", "Meg3", "Mir99ahg", "Sox2ot", "Gm20754", "Lncpint", "A330015K06Rik", "Kcnq1ot1", "2900097C17Rik", "Gm10076", "Gas5"), ]


ggplot(lncRNA_WT_TPM_CPM.df, aes(x = CPM, y = TPM)) + geom_point(size = 1, color = "#666666", alpha = 0.4 ) +
  ggtitle("4726 lncRNAs in Bulk versus scRNA in WT samples") +
  theme_classic() + scale_x_continuous(expand = c(0,0), limits = c(0, 17)) +
  scale_y_continuous(expand = c(0,0), limits = c(0,12)) + 
  xlab("log2 average CPM + 1 snRNA WT samples") +
  ylab("log2 average TPM + 1 bulk WT samples") +
  theme(plot.title = element_text(size = 15, hjust = 0.5, family = "helvetica"),
        axis.title = element_text(size = 15),
        axis.text = element_text(size = 12, color = "black")) +
  geom_abline(slope = 1, color="red", linetype="dashed", size=.5) + 
  geom_abline(slope = 1, intercept =  2 ,color="blue", linetype="dashed", size=.5) + 
  geom_abline(slope = 1, intercept =  -2 ,color="blue", linetype="dashed", size=.5) +
  geom_text_repel(data = lncRNA_WT_TPM_CPM_sub.df, aes(x = CPM, y = TPM), label = lncRNA_WT_TPM_CPM_sub.df$gene)
  
 
```

Below shows how many lncRNAs located and used for GO analysis

```{r warning=FALSE, message=FALSE}
lncRNA_WT_TPM_CPM.df$fraction <- ifelse(lncRNA_WT_TPM_CPM.df$CPM - lncRNA_WT_TPM_CPM.df$TPM > 2, "nucleus",
                                        ifelse(lncRNA_WT_TPM_CPM.df$TPM - lncRNA_WT_TPM_CPM.df$CPM > 2, "cytoplasm", "none")) 

table(lncRNA_WT_TPM_CPM.df$fraction)
```

### GO

```{r warning=FALSE, message=FALSE}
source("/data/rajewsky/home/skim/Microglia_Heppner/Codes/DroNc_AD/src/TopGO_scRNA.R")

geneUniverse <- WT_TPM_CPM$gene
genesOfInterest_nucleus <- lncRNA_WT_TPM_CPM.df[lncRNA_WT_TPM_CPM.df$fraction %in% "nucleus", ]$gene

Run_GO <- function(geneUniverse, genesOfInterest){
  # create gene list
  geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
  names(geneList) <- geneUniverse

onts = c( "MF", "BP", "CC" )
tab <- as.list(onts)
names(tab) <- onts

for(i in 1:3){

GOdata <- new("topGOdata",
                description = "GOanalysis",
                ontology = onts[i],
                allGenes = geneList,
                annot = annFUN.org,
                mapping = "org.Mm.eg.db",
                ID = "SYMBOL",
                nodeSize = 20) # nodeSize: prune the GO hierarchy less than nodeSize

# export GO ID from BP
if(i == 1){
  allGO_MF <<- genesInTerm(GOdata)
}

if(i == 2){
  allGO_BP <<- genesInTerm(GOdata)
}
if(i == 3){
  allGO_CC <<- genesInTerm(GOdata)
}
res.result1 <- runTest(GOdata, statistic = "fisher", algorithm = "elim")
res.result2 <- runTest(GOdata, statistic = "fisher", algorithm = "classic")


tab[[i]] <- cbind(data.frame(category = onts[i]), GenTable(GOdata, Fisher.elim = res.result1,
                        Fisher.classic = res.result2,
                        orderBy = "Fisher.elim" , topNodes = 30))


}

topGOResults <- plyr::rbind.fill(tab)
topGOResults.df <- as.data.frame(topGOResults)

topGOResults.df$gene_ratio <- topGOResults.df$Significant / topGOResults.df$Annotated

# modification appropriate for plot
topGOResults.df$Fisher.elim <- as.numeric(topGOResults.df$Fisher.elim)
topGOResults.df$Fisher.classic <- as.numeric(topGOResults.df$Fisher.classic)
topGOResults.df$Term <- factor(topGOResults.df$Term, levels = rev(unique(topGOResults.df$Term)))

return(topGOResults.df)
}

Nucleus_GO.df <- Run_GO(geneUniverse = geneUniverse, genesOfInterest = genesOfInterest_nucleus)

```

```{r warning=FALSE, message=FALSE}

genesOfInterest_cytoplasm <- lncRNA_WT_TPM_CPM.df[lncRNA_WT_TPM_CPM.df$fraction %in% "cytoplasm", ]$gene

Cytoplasm_GO.df <- Run_GO(geneUniverse = geneUniverse, genesOfInterest = genesOfInterest_cytoplasm)

```

```{r warning=FALSE, message=FALSE}
scRNA_TopGO_plot2 <- function(topGOResults.df){

topGOResults.df$Fisher.elim <- as.numeric(topGOResults.df$Fisher.elim)
topGOResults.df$Fisher.elim <- -log10(topGOResults.df$Fisher.elim)
topGOResults.df$Term <- factor(topGOResults.df$Term, levels = rev(unique(topGOResults.df$Term)))

ggplot(topGOResults.df, aes(x=gene_ratio,
               y=Term,
               colour=Fisher.elim,
               size=Significant)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="gene ratio", y="GO term", colour="-log10(p-value)", size="Significant") +
  theme_minimal() + theme(axis.text = element_text(size = 10))

}
```


```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=5}
g1 <- scRNA_TopGO_plot2(Nucleus_GO.df[c(61:70),])
g2 <- scRNA_TopGO_plot2(Cytoplasm_GO.df[c(61:70),])
grid.arrange(g1, g2, ncol = 2)
```