---
title: "CCI"
author: "Skim"
date: "1/13/2020"
output: 
  html_document:
    df_print: paged
    toc: true
    toc_depth: 4
    toc_float: false
    code_folding: hide
---

In this report, I tested **NicheNet** and **CellphoneDB** for **cell-cell interaction** analysis. As pre-liminary analysis, I only compared **Microglia** and **Oligodendrocytes** and compared **Ctrl** and **AD**. (Oligodendrocytes clusters (1,5,6) and microglia clusters (3, 8, 29, 34, 37)) In here, I corrected AD & ADp40KO for analysis as previous changed between two.  

```{r warning=FALSE, message=FALSE}
library(FNN)
library(nichenetr)
library(Seurat)
library(tidyverse)
library(biomaRt)
library(dplyr)
library(stringr)
library(ggplot2)
```

```{r warning=FALSE, message=FALSE}
# old resolution 0.6
#load("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200104_pre_analysis_9sets_Seurat_object.Robj")
# new resolutio 0.8 by Shirin
load("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200117_Clustering_DE_by_MAST_Seurat_object.Robj")
# data was swapped between AD and ADp40KO
#data9set.SO@meta.data$sample <- ifelse((data9set.SO$gemgroup == 1 | data9set.SO$gemgroup == 4 | data9set.SO$gemgroup == 7), "Ctrl", ifelse((data9set.SO$gemgroup == 2 | data9set.SO$gemgroup == 5 | data9set.SO$gemgroup == 8), "AD", "ADp40KO"))
# new swapped assignment
data9set.SO@meta.data$sample <- ifelse((data9set.SO$gemgroup == 1 | data9set.SO$gemgroup == 4 | data9set.SO$gemgroup == 7), "Ctrl", ifelse((data9set.SO$gemgroup == 2 | data9set.SO$gemgroup == 5 | data9set.SO$gemgroup == 8), "ADp40KO", "AD"))


data9set.SO@meta.data$sample <- factor(data9set.SO@meta.data$sample, levels = c("Ctrl", "AD", "ADp40KO"))

# sample for nichenet
data9set.SO$sample_nichenet <- as.character(data9set.SO$sample)


data9set.SO <- RenameIdents(data9set.SO, '0' = "Dentate Gyrus", '1' = "Oligo", '2' = "Neurons", '3' = "Microglia", 
                            '4' = "Astrocytes", '5' = "Oligo", '6' = "Oligo", '7' = "Inhibitory Interneurons", '8' = "Microglia", 
                            '9' = "Neuron CA2/CA3", '10' = "Dentate Gyrus", '11' = "Subiculum", '12' = "OPC", 
                            '13' = "Subiculum", '14' = "Subiculum", '15' = "Neurons", 
                            '16' = "Interneurons", '17' = "Neuron (Interneuron/subiculum)", '18' = "Neurons",
                            '19' = "Neurons", '20' = "Inhibitory Interneurons", '21' = "Inhibitory Interneurons", '22' = "Inhibitory Interneurons",
                            '23' = "Fibroblast", '24' = "Neurons", '25' = "Dentate Gyrus", '26' = "Choroid Plexus", 
                            '27' = "Neurons (Subiculum/Interneurons/ExCA)", '28' = "Cajal Rutz", '29' = "Microglia/Neuron/Fibroblast", '30' = "Inhibitory Interneurons",
                            '31' = "Dentate Gyrus", '32' = "Neurons", '33' = "Astrocytes", '34' = "Microglia",
                            '35' = "Pericytes", '36' = "Vascular Endothelial",'37' = "Microglia", '38' = "OPC",
                            '39' = "VLMC", '40' = "Neuron (subiculum/Cajal Rutz)", '41' = "OPC", '42' = "OPC", 
                            '43' = "Astrocytes", '44' = "OPC")
```


## 1. NicheNet

NicheNet can predict:  
1) which **ligands** from one cell population(**sender**) are most likely to affect **target gene** expression in an interacting cell population (**receiver**)   
2) which **specific target genes** are affected by which of these **predicted ligands**

NicheNet analysis of the following steps:  
1) Define a sender cell population and a receiver cell population / determine which genes are expressed in both populations  
2) Define a gene set of interest, these are the genes in the **receiver** cell population (e.g. genes differentially expressed upon cell-cell interaction)  
3) Define a set of potential ligands (ligands that are expressed by the **sender** cell population)  
4) Perform NicheNet ligand activity analysis (rank the potential ligands)  
5) Infer top-predicted target genes of ligands that are top-ranked in the ligand activity analysis  

I followed below link [NicheNet from Seurat](https://github.com/saeyslab/nichenetr/blob/master/vignettes/seurat_steps.md)

#### 1-1. load NicheNetâ€™s ligand-target prior model, ligand-receptor network and weighted integrated networks

```{r warning=FALSE, message=FALSE}
# ligand-target prior model
ligand_target_matrix = readRDS(url("https://zenodo.org/record/3260758/files/ligand_target_matrix.rds"))
# ligand-receptor network
lr_network = readRDS(url("https://zenodo.org/record/3260758/files/lr_network.rds"))
# weighted integrated networks
weighted_networks = readRDS(url("https://zenodo.org/record/3260758/files/weighted_networks.rds"))
weighted_networks_lr = weighted_networks$lr_sig %>% inner_join(lr_network %>% distinct(from,to), by = c("from","to"))

# convert human to mouse symbol
#lr_network = lr_network %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na()
#colnames(ligand_target_matrix) = ligand_target_matrix %>% colnames() %>% convert_human_to_mouse_symbols()
#rownames(ligand_target_matrix) = ligand_target_matrix %>% rownames() %>% convert_human_to_mouse_symbols()

#ligand_target_matrix = ligand_target_matrix %>% .[!is.na(rownames(ligand_target_matrix)), !is.na(colnames(ligand_target_matrix))]

#weighted_networks_lr = weighted_networks_lr %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na()
#head(seuratObj@assays$RNA@data@Dimnames[[1]])
```

```{r warning=FALSE, message=FALSE}
# cellphonedb data create for Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)
data9set_CP.SO <- subset(data9set.SO, subset = seurat_clusters == c(1,5,6,3,8,29,34,37))

NicheNet_run <- function(Seurat_object = data9set_CP.SO, receiver = "receiver", condition_oi = "condition_oi", condition_reference = "condition_reference", sender = "sender"){
  nichenet_output = nichenet_seuratobj_aggregate(seurat_obj = Seurat_object, 
                                               receiver = receiver, 
                                               condition_colname = "sample_nichenet", 
                                               condition_oi = condition_oi, 
                                               condition_reference = condition_reference, 
                                               sender = sender, 
                                               ligand_target_matrix = ligand_target_matrix, 
                                               lr_network = lr_network, 
                                               weighted_networks = weighted_networks, 
                                               organism = "mouse", 
                                               verbose = FALSE)
  return(nichenet_output)
}

# AD vs Ctrl (R = O, S = MG)
nichenet_output = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Oligo", condition_oi = "AD",  condition_reference = "Ctrl", sender = "Microglia")

# AD vs Ctrl (R = MG, S = O)
nichenet_output2 = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Microglia", condition_oi = "AD", condition_reference = "Ctrl", sender = "Oligo" )

# ADp40KO vs Ctrl (R = O, S = MG)
nichenet_output3 = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Oligo", condition_oi = "ADp40KO",  condition_reference = "Ctrl", sender = "Microglia")

# ADp40KO vs Ctrl (R = MG, S = O)
nichenet_output4 = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Microglia", condition_oi = "ADp40KO", condition_reference = "Ctrl", sender = "Oligo" )

# ADp40KO vs AD (R = O, S = MG)
#nichenet_output5 = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Oligo", condition_oi = "ADp40KO",  condition_reference = "AD", sender = "Microglia")

# ADp40KO vs AD (R = MG, S = O)
#nichenet_output6 = NicheNet_run(Seurat_object = data9set_CP.SO, receiver = "Microglia", condition_oi = "ADp40KO", condition_reference = "AD", sender = "Oligo" )
```

#### 1-2. AD vs Ctrl (receiver = Oligo, sender = Microglia)

```{r warning=FALSE, message=FALSE}
#list(nichenet_output2)
#nichenet_output$ligand_activity_target_heatmap
#nichenet_output$ligand_target_heatmap
nichenet_output$ligand_receptor_heatmap

#FeaturePlot(data9set.SO, features = c("Vim", "Actb", "App", "Ccl5"), pt.size = 0.1)
```

#### 1-2. AD vs Ctrl (receiver = Microglia, sender = Oligo)

```{r warning=FALSE, message=FALSE}
#list(nichenet_output2)
#nichenet_output$ligand_activity_target_heatmap
#nichenet_output$ligand_target_heatmap
nichenet_output2$ligand_receptor_heatmap

#FeaturePlot(data9set.SO, features = c("Vim", "Actb", "App", "Ccl5"), pt.size = 0.1)
```

#### 1-3. ADp40KO vs Ctrl (receiver = Oligo, sender = Microglia)

```{r warning=FALSE, message=FALSE}
#list(nichenet_output2)
#nichenet_output$ligand_activity_target_heatmap
#nichenet_output$ligand_target_heatmap
nichenet_output3$ligand_receptor_heatmap

#FeaturePlot(data9set.SO, features = c("Vim", "Actb", "App", "Ccl5"), pt.size = 0.1)
```

#### 1-4. ADp40KO vs Ctrl (receiver = Microglia, sender = Oligo)

```{r warning=FALSE, message=FALSE}
#list(nichenet_output2)
#nichenet_output$ligand_activity_target_heatmap
#nichenet_output$ligand_target_heatmap
nichenet_output4$ligand_receptor_heatmap

#FeaturePlot(data9set.SO, features = c("Vim", "Actb", "App", "Ccl5"), pt.size = 0.1)
```

Conclusion: It seems Adipoq, Sele, Nms (ligands) and Casr, Trhr, Ccr3 (receptors) are commonly expressed in both Microglia and Oligodendrocytes. When comparing (ADp40KO vs AD), there was no highly likely active targets found for top ligands

## 2. CellphoneDB

CellphoneDB following steps:  
1) change mouse gene symbol to human symbol and extract count data and meta file from Seurat  
2) Run cellphoneDB statistical analysis (--iterations=10000)  
3) Analyze data (pvalue, meanvalue, sigmeanvalue files)

I followed below link: [cellphoneDB](https://www.biorxiv.org/content/10.1101/680926v1.full.pdf)

#### 2-1. cellphonedb data create (Ctrl)

In here, I only look Ctrl populations

```{r warning=FALSE, message=FALSE}
gene.df <- data.frame(Gene = data9set.SO@assays$RNA@counts@Dimnames[1], stringsAsFactors = FALSE)
colnames(gene.df) <- "Gene"
write.table(gene.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_GENE_DF")

mart <- useDataset("mmusculus_gene_ensembl", mart=useMart("ensembl"))

bm <- getBM(attributes=c("ensembl_gene_id", "hsapiens_homolog_associated_gene_name"),
            filters = 'mgi_symbol',
            values = data9set.SO@assays$RNA@counts@Dimnames[1],
            mart=mart)

bm <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_GENE_BM.df", sep = "\t")

bm2 <- getBM(attributes=c("ensembl_gene_id", "mgi_symbol"),
             filters = 'mgi_symbol',
             values = rownames(data9set_CP2.df),
             mart=mart)


bm2.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_GENE_BM2.df", sep = "\t")
# cellphonedb data create for Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)
# And subset Ctrl and AD
#data9set_CP2.SO <- subset(data9set_CP.SO, subset = sample == c("Ctrl"))

## cell type
#data9set_CP2.SO@meta.data$cell_type <- ifelse((data9set_CP2.SO$seurat_clusters == 1 | data9set_CP2.SO$seurat_clusters == 5 | data9set_CP2.SO$seurat_clusters == 6), "Oligo", "Microglia")

#data9set_CP2.df <- as.data.frame(as.matrix(data9set_CP2.SO@assays$RNA@counts))

# normalize
#data9set_CP2.df <- as.data.frame(apply(data9set_CP2.df, 2, function(x) (x/sum(x))*10000))

# change mouse to human gene symbols
#mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))

#gene.df <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol'), 
#      filters = 'hgnc_symbol', 
#      values = rownames(data9set_CP2.df), 
#      mart = mart)

#gene.df$hgnc_symbol <- str_to_title(gene.df$hgnc_symbol)


#data9set_CP2.df$hgnc_symbol <- rownames(data9set_CP2.df)
#data9set_CP2.df <- inner_join(gene.df, data9set_CP2.df, by = "hgnc_symbol")

# Assign ENSEMBL to rownames
#rownames(data9set_CP2.df) <- data9set_CP2.df[, 1]
# remove 1st and 2nd columns
#data9set_CP2.df <- data9set_CP2.df[, -c(1:2)]


#setdiff(colnames(data9set_CP2.df), rownames(data9set_CP.SO@meta.data))
#tail(colnames(data9set_CP2.df))
#tail(rownames(data9set_CP.SO@meta.data))

# save 
# using check.names=FALSE otherwise dash(-) will change to dot(.) which will cause error on cellphonedb

#write.table(data.frame("Genes"=rownames(data9set_CP2.df), data9set_CP2.df, check.names=FALSE), "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_Ctrl.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data create
#data9set_CP2_meta.df <- data.frame(Cell = rownames(data9set_CP2.SO@meta.data), cell_type = data9set_CP2.SO@meta.data$seurat_clusters)
#write.table(data9set_CP2_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_Ctrl.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data for cell type
#data9set_CP2_meta.df <- data.frame(Cell = rownames(data9set_CP2.SO@meta.data), cell_type = data9set_CP2.SO@meta.data$cell_type)
#write.table(data9set_CP2_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_cell_type_Ctrl.txt", sep = "\t", quote=FALSE, row.names = FALSE)

```

#### 2-2. cellphonedb analysis (Ctrl, Cell type)

In y-axis, left is ligand while right is receptor. In x-axis, first row is sender cell type while second row is receiver cell type. 

```{r warning=FALSE, message=FALSE, fig.width= 7, fig.height= 6}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl_cell_type/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl_cell_type/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl_cell_type/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:16)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:15)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:7)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:7)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, Microglia.Microglia:Oligo.Oligo)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, Microglia.Microglia:Oligo.Oligo)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)


# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > .5), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- strsplit(test2.df$sigmean, '[.]')[[i]][1]
  test2.df[i, "receiver"] <- strsplit(test2.df$sigmean, '[.]')[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), 
                          axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + facet_grid(.~sender, scales="free", space = "free", switch = "x")

```

#### 2-3. cellphonedb analysis (Ctrl, Cluster)

significant means > 2 in here. Each cluster, Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)

```{r warning=FALSE, message=FALSE, fig.width= 15, fig.height= 12}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_Ctrl/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:76)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:75)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:68)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:68)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, X1.1:X37.37)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, X1.1:X37.37)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)

# remove self clustering Oligo (1,5,6) and microglia (3, 8, 29, 34, 37)
test2.df <- test2.df[which(!test2.df$sigmean == "X1.1" & !test2.df$sigmean == "X5.5" & !test2.df$sigmean == "X6.6" & !test2.df$sigmean == "X3.3" & !test2.df$sigmean == "X8.8" & !test2.df$sigmean == "X29.29" & !test2.df$sigmean == "X34.34" & !test2.df$sigmean == "X37.37"), ] 

# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > 2), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- paste0("cluster", str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][1])
  test2.df[i, "receiver"] <- str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + 
  #facet_wrap(~sender, switch = "x", scales = "free_x", nrow = 1)
  facet_grid(.~sender, scales="free", space = "free", switch = "x")

```


#### 2-4. cellphonedb data create (AD)

In here, I only look AD cell types

```{r warning=FALSE, message=FALSE}
# cellphonedb data create for Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)
# And subset Ctrl and AD
#data9set_CP3.SO <- subset(data9set_CP.SO, subset = sample == c("AD"))

## cell type
#data9set_CP3.SO@meta.data$cell_type <- ifelse((data9set_CP3.SO$seurat_clusters == 1 | data9set_CP3.SO$seurat_clusters == 5 | data9set_CP3.SO$seurat_clusters == 6), "Oligo", "Microglia")

#data9set_CP3.df <- as.data.frame(as.matrix(data9set_CP3.SO@assays$RNA@counts))

# normalize
#data9set_CP3.df <- as.data.frame(apply(data9set_CP3.df, 2, function(x) (x/sum(x))*10000))

# change mouse to human gene symbols
#mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))

#gene.df <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol'), 
#      filters = 'hgnc_symbol', 
#     values = rownames(data9set_CP3.df), 
#      mart = mart)

#gene.df$hgnc_symbol <- str_to_title(gene.df$hgnc_symbol)


#data9set_CP3.df$hgnc_symbol <- rownames(data9set_CP3.df)
#data9set_CP3.df <- inner_join(gene.df, data9set_CP3.df, by = "hgnc_symbol")

# Assign ENSEMBL to rownames
#rownames(data9set_CP3.df) <- data9set_CP3.df[, 1]
# remove 1st and 2nd columns
#data9set_CP3.df <- data9set_CP3.df[, -c(1:2)]


#setdiff(colnames(data9set_CP2.df), rownames(data9set_CP.SO@meta.data))
#tail(colnames(data9set_CP2.df))
#tail(rownames(data9set_CP.SO@meta.data))

# save 
# using check.names=FALSE otherwise dash(-) will change to dot(.) which will cause error on cellphonedb

#write.table(data.frame("Genes"=rownames(data9set_CP3.df), data9set_CP3.df, check.names=FALSE), "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_AD.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data create
#data9set_CP3_meta.df <- data.frame(Cell = rownames(data9set_CP3.SO@meta.data), cell_type = data9set_CP3.SO@meta.data$seurat_clusters)
#write.table(data9set_CP3_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_AD.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data cell type create
#data9set_CP3_meta.df <- data.frame(Cell = rownames(data9set_CP3.SO@meta.data), cell_type = data9set_CP3.SO@meta.data$cell_type)
#write.table(data9set_CP3_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_cell_type_AD.txt", sep = "\t", quote=FALSE, row.names = FALSE)

```

#### 2-5. cellphonedb analysis (AD, cell type)

In y-axis, left is ligand while right is receptor. In x-axis, first row is sender cell type while second row is receiver cell type.

```{r warning=FALSE, message=FALSE, fig.width= 7, fig.height= 6}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD_cell_type/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD_cell_type/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD_cell_type/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:16)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:15)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:7)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:7)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, Microglia.Microglia:Oligo.Oligo)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, Microglia.Microglia:Oligo.Oligo)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)


# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > .5), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- strsplit(test2.df$sigmean, '[.]')[[i]][1]
  test2.df[i, "receiver"] <- strsplit(test2.df$sigmean, '[.]')[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), 
                          axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + facet_grid(.~sender, scales="free", space = "free", switch = "x")

```

#### 2-6. cellphonedb analysis (AD, cluster)

significant means > 2 in here. Each cluster, Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)

```{r warning=FALSE, message=FALSE, fig.width= 15, fig.height= 10}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_AD/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:76)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:75)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:68)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:68)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, X1.1:X37.37)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, X1.1:X37.37)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)

# remove self clustering Oligo (1,5,6) and microglia (3, 8, 29, 34, 37)
test2.df <- test2.df[which(!test2.df$sigmean == "X1.1" & !test2.df$sigmean == "X5.5" & !test2.df$sigmean == "X6.6" & !test2.df$sigmean == "X3.3" & !test2.df$sigmean == "X8.8" & !test2.df$sigmean == "X29.29" & !test2.df$sigmean == "X34.34" & !test2.df$sigmean == "X37.37"), ] 

# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > .5), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- paste0("cluster", str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][1])
  test2.df[i, "receiver"] <- str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + 
  #facet_wrap(~sender, switch = "x", scales = "free_x", nrow = 1)
  facet_grid(.~sender, scales="free", space = "free", switch = "x")

```

#### 2-7. cellphonedb data create (ADp40KO)

In here, I only look ADp40KO cell types

```{r warning=FALSE, message=FALSE}
# cellphonedb data create for Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)
# And subset Ctrl and AD
#data9set_CP4.SO <- subset(data9set_CP.SO, subset = sample == c("ADp40KO"))

## cell type
#data9set_CP4.SO@meta.data$cell_type <- ifelse((data9set_CP4.SO$seurat_clusters == 1 | data9set_CP4.SO$seurat_clusters == 5 | data9set_CP4.SO$seurat_clusters == 6), "Oligo", "Microglia")

#data9set_CP4.df <- as.data.frame(as.matrix(data9set_CP4.SO@assays$RNA@counts))

# normalize
#data9set_CP4.df <- as.data.frame(apply(data9set_CP4.df, 2, function(x) (x/sum(x))*10000))

# change mouse to human gene symbols
#mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))

#gene.df <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol'), 
#      filters = 'hgnc_symbol', 
#     values = rownames(data9set_CP4.df), 
#      mart = mart)

#gene.df$hgnc_symbol <- str_to_title(gene.df$hgnc_symbol)


#data9set_CP4.df$hgnc_symbol <- rownames(data9set_CP4.df)
#data9set_CP4.df <- inner_join(gene.df, data9set_CP4.df, by = "hgnc_symbol")

# Assign ENSEMBL to rownames
#rownames(data9set_CP4.df) <- data9set_CP4.df[, 1]
# remove 1st and 2nd columns
#data9set_CP4.df <- data9set_CP4.df[, -c(1:2)]


#setdiff(colnames(data9set_CP2.df), rownames(data9set_CP.SO@meta.data))
#tail(colnames(data9set_CP2.df))
#tail(rownames(data9set_CP.SO@meta.data))

# save 
# using check.names=FALSE otherwise dash(-) will change to dot(.) which will cause error on cellphonedb

#write.table(data.frame("Genes"=rownames(data9set_CP4.df), data9set_CP4.df, check.names=FALSE), "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_ADp40KO.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data create
#data9set_CP4_meta.df <- data.frame(Cell = rownames(data9set_CP4.SO@meta.data), cell_type = data9set_CP4.SO@meta.data$seurat_clusters)
#write.table(data9set_CP4_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_ADp40KO.txt", sep = "\t", quote=FALSE, row.names = FALSE)

# cellphonedb meta data cell type create
#data9set_CP4_meta.df <- data.frame(Cell = rownames(data9set_CP4.SO@meta.data), cell_type = data9set_CP4.SO@meta.data$cell_type)
#write.table(data9set_CP4_meta.df, "/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200113_CCI_test_cellphoneDB_Oligo_MG_meta_cell_type_ADp40KO.txt", sep = "\t", quote=FALSE, row.names = FALSE)

```

#### 2-8. cellphonedb analysis (ADp40KO, cell type)

In y-axis, left is ligand while right is receptor. In x-axis, first row is sender cell type while second row is receiver cell type.

```{r warning=FALSE, message=FALSE, fig.width= 7, fig.height= 6}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO_cell_type/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO_cell_type/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO_cell_type/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:16)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:15)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:7)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:7)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, Microglia.Microglia:Oligo.Oligo)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, Microglia.Microglia:Oligo.Oligo)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)


# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > .5), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- strsplit(test2.df$sigmean, '[.]')[[i]][1]
  test2.df[i, "receiver"] <- strsplit(test2.df$sigmean, '[.]')[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), 
                          axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + facet_grid(.~sender, scales="free", space = "free", switch = "x")

```

#### 2-9. cellphonedb analysis (ADp40KO, cluster)

significant means > 2 in here. Each cluster, Oligodendrocytes (1,5,6) and microglia (3, 8, 29, 34, 37)

```{r warning=FALSE, message=FALSE, fig.width= 15, fig.height= 11}
pvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO/pvalues.txt", sep = "\t", header = TRUE)
meanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO/means.txt", sep = "\t", header = TRUE)
sigmeanvalue.df <- read.table("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/out/cellphoneDB_Oligo_MG_ADp40KO/significant_means.txt", sep = "\t", header = TRUE)

# sigmeanvalue modification
sigmeanvalue.df[is.na(sigmeanvalue.df)] <- 0

sigmeanvalue_test.df <- data.frame(sigmeanvalue.df[, c(2, 5,6, 13:76)], na.rm=TRUE)

sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

# pvalue modification
pvalue_test.df <- data.frame(pvalue.df[, c(2, 5,6, 12:75)], na.rm=TRUE)
pvalue_test.df <- pvalue_test.df[which(pvalue_test.df$interacting_pair %in% sigmeanvalue_test.df$interacting_pair & pvalue_test.df$gene_a %in% sigmeanvalue_test.df$gene_a & pvalue_test.df$gene_b %in% sigmeanvalue_test.df$gene_b), ]

# remove overapping information
sigmeanvalue_test.df <- sigmeanvalue_test.df[, c(1,4:68)]
sigmeanvalue_test.df <- sigmeanvalue_test.df %>% distinct()

pvalue_test.df <- pvalue_test.df[, c(1,4:68)]
pvalue_test.df <- pvalue_test.df %>% distinct()


sigmeanvalue_test2.df <- sigmeanvalue_test.df %>% gather(sigmean, sigmeanvalue, X1.1:X37.37)
pvalue_test2.df <- pvalue_test.df %>% gather(p, pvalue, X1.1:X37.37)


sigmeanvalue_test2.df$ID <- paste0(sigmeanvalue_test2.df$interacting_pair, sigmeanvalue_test2.df$sigmean)
pvalue_test2.df$ID <- paste0(pvalue_test2.df$interacting_pair, pvalue_test2.df$p)

test2.df <- merge(sigmeanvalue_test2.df, pvalue_test2.df, by = "ID")

test2.df$logpvalue <- -log10(test2.df$pvalue)

# remove self clustering Oligo (1,5,6) and microglia (3, 8, 29, 34, 37)
test2.df <- test2.df[which(!test2.df$sigmean == "X1.1" & !test2.df$sigmean == "X5.5" & !test2.df$sigmean == "X6.6" & !test2.df$sigmean == "X3.3" & !test2.df$sigmean == "X8.8" & !test2.df$sigmean == "X29.29" & !test2.df$sigmean == "X34.34" & !test2.df$sigmean == "X37.37"), ] 

# remove 0 in sigmean
test2.df <- test2.df[which(test2.df$sigmeanvalue > 0), ]
# assign Infinity in logpvalue as 3
test2.df$logpvalue[is.infinite(test2.df$logpvalue)] <- 3
# subset sigmeanvalue > 1
test2.df <- test2.df[which(test2.df$sigmeanvalue > .5), ]

for(i in 1:nrow(test2.df)){
  test2.df[i, "sender"] <- paste0("cluster", str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][1])
  test2.df[i, "receiver"] <- str_split(sub(".", "", test2.df$sigmean), "\\.")[[i]][2]
}

ggplot(test2.df, aes(x=receiver,
               y=interacting_pair.x,
               colour=sigmeanvalue,
               size=logpvalue)) +
        geom_point() +
        expand_limits(x=0) +
        labs(x="cluster", y="Signaling", colour="mean expression", size="-log10(p value)") +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.text.x =  element_text(size = 10), 
                          panel.margin = unit(0, "lines"), 
         strip.background = element_blank()) + 
  #facet_wrap(~sender, switch = "x", scales = "free_x", nrow = 1)
  facet_grid(.~sender, scales="free", space = "free", switch = "x")

```

Conclusion: TTR_DDR1, GPR37_PSAP, CSF1R_CSF1, CSF1_SIRPA, CADM1_CADM1 are most changed pairs

```{r warning=FALSE, message=FALSE, fig.width= 12, fig.height= 7}
sessionInfo()
```
