---
title: "DE_topGO_GSEA"
author: "Skim"
date: "1/8/2020"
output: html_document
---

[GSEA ref](https://stephenturner.github.io/deseq-to-fgsea/#non-human_organisms)
[GO ref](https://www.huber.embl.de/users/klaus/Teaching/DESeq2Predoc2014.html#gene-ontology-enrichment-analysis)
```{r warning=FALSE, message=FALSE}
Sys.setenv(PYTHONPATH="/data/local/rajewsky/shared_libs/python")
library(reticulate)
#use_python("/usr/bin/python3", required = TRUE)
library(Seurat)
library(ggplot2)
library(dplyr)
library(MAST)
library(edgeR)
library(DT)
library(gridExtra)
library(nichenetr)
library(tidyverse)
library(fgsea)
library(MultiAssayExperiment)
.libPaths(c("/data/murphy/shared_libs/R", "/usr/local/lib/R/site-librar","/usr/local/lib/R/library","/usr/lib64/R/library","/usr/share/R/library", .libPaths() ))
#Rscript -e "library(rmarkdown); rmarkdown::render('/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200108_DE_GO_GSEA.Rmd')"
```

```{r warning=FALSE, message=FALSE}
load("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/20200104_pre_analysis_9sets_Seurat_object.Robj")
```

```{r warning=FALSE, message=FALSE, fig.width= 10, fig.height= 6}
data9set.SO <- RenameIdents(data9set.SO, '0' = "dentate", '1' = "Mature oligo", '2' = "CA1 subiculum", '3' = "Microglia", 
                            '4' = "Mature oligo", '5' = "Astrocytes", '6' = "interneuron", '7' = "CA2/CA3", '8' = "subiculum", 
                            '9' = "OPC", '10' = "CA1 subiculum", '11' = "interneuron/subiculum", '12' = "interneuron/subiculum", 
                            '13' = "interneuron", '14' = "interneuron/subiculum", '15' = "interneuron/subiculum", 
                            '16' = "interneuron/subiculum", '17' = "interneuron/subiculum", '18' = "interneuron",
                            '19' = "CA2/CA3", '20' = "Fibroblast", '21' = "CA1 subiculum/interneuron", '22' = "dentate",
                            '23' = "Chorid plexus", '24' = "CA2/CA3", '25' = "Cajal-Retzius", '26' = "dentate", 
                            '27' = "Astrocytes", '28' = "CA2/CA3", '29' = "Microglia", '30' = "Endothelial", '31' = "Endothelial",
                            '32' = "Microglia", '33' = "polydendrocytes", '34' = "OPC")

data9set.SO@active.ident <- factor(data9set.SO@active.ident, levels = c("dentate", "CA1 subiculum", "CA1 subiculum/interneuron", "CA2/CA3",
                                                                        "interneuron", "interneuron/subiculum", "subiculum", "Astrocytes",
                                                                        "Mature oligo", "polydendrocytes", "OPC", "Microglia",
                                                                        "Endothelial", "Cajal-Retzius", "Chorid plexus", "Fibroblast"))
```


## Microglia analysis

```{r warning=FALSE, message=FALSE}

data9set.SO_MG <- subset(data9set.SO, subset = seurat_clusters == c(3, 29, 32))

AD_Ctrl_MG <- FindMarkers(data9set.SO_MG, ident.1 =  "AD", ident.2 = "Ctrl", 
                             group.by = "sample", test.use = "MAST", logfc.threshold = 0)

###############################
# MASTcpmDetRate
# ref: https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_MASTcpmDetRate.R

data(maits, package='MAST')
dim(maits$expressionmat)
head(maits$cdat)
head(maits$fdat, 10)

dge <- DGEList(data9set.SO_MG@assays$RNA@counts)
dge <- edgeR::calcNormFactors(dge)
cpms <- cpm(dge)
sca <- FromMatrix(exprsArray = log2(cpms + 1), cData = data.frame(wellKey = names(grp), grp = grp, cdr = cdr))


data(miniACC)
a1 <- experiments(miniACC)
a1$RNASeq2GeneNorm

colData(miniACC)
```

## GSEA

I downloaded **Hallmark gene set** from **MSigDB** (http://software.broadinstitute.org/gsea/msigdb/index.jsp). Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying overlaps between gene sets in other MSigDB collections and retaining genes that display coordinate expression. I also downloaded **Reactome Pathways Gene Set** from **reactome** (https://reactome.org/download-data)
To do, I refered below discussion (https://github.com/ctlab/fgsea/issues/50) 

```{r warning=FALSE, message=FALSE}
# Creating rnk file
AD_Ctrl_MG$Gene <- rownames(AD_Ctrl_MG)
AD_Ctrl_MG$fcsign <- sign(AD_Ctrl_MG$avg_logFC)
AD_Ctrl_MG$logP = -log10(AD_Ctrl_MG$p_val)
AD_Ctrl_MG$metric= AD_Ctrl_MG$logP/AD_Ctrl_MG$fcsign

AD_Ctrl_MG_metric <- AD_Ctrl_MG[,c("Gene", "metric")]
AD_Ctrl_MG_metric$Gene <- toupper(AD_Ctrl_MG_metric$Gene)

pathways.hallmark <- gmtPathways("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/ext_files/ReactomePathways.gmt")
pathways.hallmark2 <- gmtPathways("/data/rajewsky/home/skim/Microglia_Heppner/201912_10X_9sets/R_Scripts/ext_files/h.all.v7.0.symbols.gmt")

ranks <- tibble::deframe(AD_Ctrl_MG_metric)
fgseaRes <- fgsea(pathways.hallmark, ranks, minSize=15, maxSize=500, nperm=1000)
fgseaRes2 <- fgsea(pathways.hallmark2, ranks, minSize=15, maxSize=500, nperm=1000)


topUp <- fgseaRes %>% 
    filter(ES > 0) %>% 
    top_n(10, wt=-padj)
topDown <- fgseaRes %>% 
    filter(ES < 0) %>% 
    top_n(10, wt=-padj)
topPathways <- bind_rows(topUp, topDown) %>% 
    arrange(-ES)


datatable(topPathways, options = list(pageLength = 5, dom = 'tip'))

barplot(sort(ranks, decreasing = T))

```

```{r warning=FALSE, message=FALSE, fig.height= 10, fig.width= 10}
fgseaResTidy <- fgseaRes %>% as_tibble() %>% arrange(desc(NES))



fgseaResTidy_1 <- fgseaResTidy[c(1:20), ]
fgseaResTidy_2 <- fgseaResTidy[c((nrow(fgseaResTidy)-19):nrow(fgseaResTidy)), ]


fgseaResTidy_3 <- rbind(fgseaResTidy_1, fgseaResTidy_2) 
fgseaResTidy_3 <- fgseaResTidy_3 %>% as_tibble() %>% arrange(desc(NES))

ggplot(fgseaResTidy_3, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score") + 
  theme_minimal()
```

```{r warning=FALSE, message=FALSE, fig.height= 10, fig.width= 10}
fgseaResTidy2 <- fgseaRes2 %>% as_tibble() %>% arrange(desc(NES))
ggplot(fgseaResTidy2, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score") + 
  theme_minimal()

plotEnrichment(pathways.hallmark2[["HALLMARK_IL2_STAT5_SIGNALING"]], ranks)
```
## test mouse genome

```{r warning=FALSE, message=FALSE, fig.width= 5, fig.height=4}
library(biomaRt)
mart <- useDataset("mmusculus_gene_ensembl", mart=useMart("ensembl"))

bm <- getBM(attributes=c("ensembl_gene_id", "hsapiens_homolog_associated_gene_name"),  
            filters = 'mgi_symbol',
            values = rownames(data9set_MG.marker0),
            mart=mart)

bm <- bm %>% distinct() %>% as_tibble() %>% na_if("") %>% na.omit() %>% as.data.frame()

bm.df <- data.frame(ensembl_gene_id= bm$ensembl_gene_id, hsapiens_homolog_associated_gene_name = bm$hsapiens_homolog_associated_gene_name, stringsAsFactors = FALSE)


bm2 <- getBM(attributes=c("ensembl_gene_id", "mgi_symbol"),  
            filters = 'mgi_symbol',
            values = rownames(data9set_MG.marker0),
            mart=mart)

bm2.df <- as.data.frame(bm2)

colnames(bm2.df) %in% colnames(bm.df)

names(bm.df)

names(bm.df) <- gsub("\\s"," ",names(bm.df))
names(bm.df) <- names(bm.df) %>% stringr::str_replace_all("\\s","_")


total <- merge(bm.df,bm2.df, by = "ensembl_gene_id")

capFirst <- function(s) { 

    paste(toupper(substring(s, 1, 1)), substring(s, 2), sep = "")} 

total$mgi_symbol <- toupper(total$mgi_symbol) 

total$sanity <- ifelse(total$hsapiens_homolog_associated_gene_name == total$mgi_symbol, "same", "diff")


###########
# run GSEA
total2 <- data.frame(Gene = total$mgi_symbol, HGene = total$hsapiens_homolog_associated_gene_name)


data9set_MG.marker0_HUMAN <- merge(AD_Ctrl_MG_metric, total2, by = "Gene")


data9set_MG.marker0_HUMAN <- right_join(AD_Ctrl_MG_metric, total2, by = "Gene")

data9set_MG.marker0_HUMAN <- data9set_MG.marker0_HUMAN[, c(3,2)]

ranks <- tibble::deframe(data9set_MG.marker0_HUMAN)

fgseaRes <- fgsea(pathways.hallmark, ranks, minSize=10, maxSize=500, nperm=100000)
fgseaRes2 <- fgsea(pathways.hallmark2, ranks, minSize=15, maxSize=500, nperm=100000)
```
## GO


